applications/serial_lte_modem/src/nrfcloud/slm_at_nrfcloud.c

```diff
@@ -316,6 +322,38 @@ static int do_cloud_send_msg(const char *message, int len)
 	return err;
 }
 
+static int do_cloud_send_generic_msg(const char *message, int len)
```

Comment:
do not use REST, use the existing MQTT connection.

Reply:
```
static int do_cloud_send_generic_msg(const char *message, int len)
{
	int err;
	char *msg_str;
	NRF_CLOUD_OBJ_JSON_DEFINE(msg_obj);
	struct nrf_cloud_tx_data mqtt_msg = {
		.qos = MQTT_QOS_1_AT_LEAST_ONCE,
		.topic_type = NRF_CLOUD_TOPIC_MESSAGE,
		.obj = &msg_obj
	};

	err = nrf_cloud_obj_init(&msg_obj);
	if (err) {
		LOG_ERR("Failed to initialize message object, error: %d", err);
		return -ENOMEM;
	}

	/* Allocate a buffer to create a NULL terminated string */
	msg_str = k_malloc(len + 1);
	if (!msg_str){
		LOG_WRN("Unable to allocate buffer");
		err = -ENOMEM;
		goto cleanup;
	}
	memcpy(msg_str, message, len);
	msg_str[len] = 0;

	/* Build a simple JSON message with "msg" as the key and the
	 * provided message as the value
	 */
	err = nrf_cloud_obj_str_add(&msg_obj, "msg", msg_str, false);
	if (err) {
		LOG_ERR("Failed to create message, error: %d", err);
		err = -ENOMEM;
		goto cleanup;
	}

	/* It is safe to free the message string since it has been added to the object */
	k_free(msg_str);
	msg_str = NULL;

	/* Send message */
	err = nrf_cloud_send(&mqtt_msg);
	if (err) {
		LOG_ERR("nrf_cloud_send failed, error: %d", err);
	}

cleanup:
	k_free(msg_str);
	(void)nrf_cloud_obj_free(&msg_obj);

	return err;
}
```

Reply:
it would be nice if you could ensure that the `const char *message` parameter is NULL terminated... that would make things cleaner in this function.  then you could remove the `len` parameter and the malloc.